src/tokens/hooks.ts

```diff
@@ -4,19 +4,73 @@ import useSelector from 'src/redux/useSelector'
 import {
   tokensByAddressSelector,
   tokensByCurrencySelector,
-  tokensListSelector,
+  tokensByIdSelectorWrapper,
+  tokensListSelectorWrapper,
+  totalTokenBalanceSelectorWrapper,
+  tokensWithUsdValueSelectorWrapper,
+  tokensListWithAddressSelector,
 } from 'src/tokens/selectors'
 import { convertLocalToTokenAmount, convertTokenToLocalAmount } from 'src/tokens/utils'
 import { Currency } from 'src/utils/currencies'
+import { NetworkId } from 'src/transactions/types'
+import { TIME_UNTIL_TOKEN_INFO_BECOMES_STALE, TOKEN_MIN_AMOUNT } from 'src/config'
+import networkConfig from 'src/web3/networkConfig'
 
-export function useTokenInfo(tokenAddress?: string | null) {
+/**
+ * @deprecated use useTokenInfo and select using tokenId
+ */
+export function useTokenInfoByAddress(tokenAddress?: string | null) {
   // TODO: Make this work with native tokens lacking a tokenAddress
   const tokens = useSelector(tokensByAddressSelector)
   return tokenAddress ? tokens[tokenAddress] : undefined
 }
 
-export function useTokenInfoBySymbol(symbol: string) {
-  const tokens = useSelector(tokensListSelector)
+export function useTokensWithUsdValue(networkIds: NetworkId[]) {
+  return useSelector(tokensWithUsdValueSelectorWrapper(networkIds))
+}
+
+export function useTotalTokenBalance(networkIds: NetworkId[]) {
+  return useSelector(totalTokenBalanceSelectorWrapper(networkIds))
+}
```

Comment:
I wonder if this should take networkIds or just filter by network ids supported for balance. One the one hand its nice that all of the functions here take a networkId, but is there a use case for when we need use different network ids for total token balance in different parts of the app? Also, I guess we'd need some wrapper for getting tokens eligible for send / swap / CICO which can also wrap the statsig logic and doesn't need to take network ids

Reply:
Yeah, I think this is probably a good idea -- I've updated the implementation. We'll still need to get the supported networks from Statsig directly from the components, because the components that use `useTotalTokenBalance` _also_ use other hooks that _do_ require a list of network IDs (e.g. `useTokensWithUsdValue`). Despite this, I think there's no harm in having the Statsig logic within `useTotalTokenBalance` as, it's very possible we use this elsewhere in the app, and this lowers the risk of passing the "wrong" network IDs.

Reply:
I've also added this treatment to `useTokensWithTokenBalance` as well, since I don't imagine we'll want _individual_ token balances to appear for all chains but not be reflected in _total_ balance, or vice versa.